File: CS/LocalizeApplication/Helpers/LocalizeEnumHelper.cs
using System;
using LocalizeApplication.Resources.Localization;
using LocalizeApplication.Model;
using System.Reflection;
using System.Data;

namespace LocalizeApplication.Helpers
{
    public static class LocalizeEnumHelper
    {
        public static string GetLocalizedString(NumberTypes numbertype)
        {
            switch (numbertype)
            {
                case NumberTypes.Mobile: return Resources.Localization.ApplicationStrings.NumberType_Mobile;
                case NumberTypes.Home: return Resources.Localization.ApplicationStrings.NumberType_Home;
                case NumberTypes.Work: return Resources.Localization.ApplicationStrings.NumberType_Work;
                case NumberTypes.Unknown: return Resources.Localization.ApplicationStrings.NumberType_Unknown;

                default: throw new ArgumentOutOfRangeException("numbertype");
            }
        }
        public static string GetIcon(CallTypes calltype)
        {
            switch (calltype)
            {
                case CallTypes.Incoming: return "→";
                case CallTypes.Outgoing: return "←";

                default: throw new ArgumentOutOfRangeException("numbertype");
            }
        }
    }
}

File: VB/LocalizeApplication/Helpers/LocalizeEnumHelper.vb
Imports System
Imports LocalizeApplication.Resources.Localization
Imports LocalizeApplication.Model
Imports System.Reflection
Imports System.Data

Namespace LocalizeApplication.Helpers

    Public Module LocalizeEnumHelper

        Public Function GetLocalizedString(ByVal numbertype As NumberTypes) As String
            Select Case numbertype
                Case NumberTypes.Mobile
                    Return ApplicationStrings.NumberType_Mobile
                Case NumberTypes.Home
                    Return ApplicationStrings.NumberType_Home
                Case NumberTypes.Work
                    Return ApplicationStrings.NumberType_Work
                Case NumberTypes.Unknown
                    Return ApplicationStrings.NumberType_Unknown
                Case Else
                    Throw New ArgumentOutOfRangeException("numbertype")
            End Select
        End Function

        Public Function GetIcon(ByVal calltype As CallTypes) As String
            Select Case calltype
                Case CallTypes.Incoming
                    Return "→"
                Case CallTypes.Outgoing
                    Return "←"
                Case Else
                    Throw New ArgumentOutOfRangeException("numbertype")
            End Select
        End Function
    End Module
End Namespace
